### PR TITLE
fix: make grid connector request content update on missing data

### PR DESCRIPTION
**Pull request by tomivirkki**
Fixes https://github.com/vaadin/flow-components/issues/2389

### CODE REVIEW

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java [hunk 1]
```diff
@@ -693,6 +693,23 @@ public void gridWithFiltering() {
         waitUntil(driver -> grid.getCell(0, 0).getText().contains("Person 6"));
     }
 
+    @Test
+    public void gridWithFiltering_scrolledToEnd() {
```
Comment (sissbruecker) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java:697]: The test name could use an expectation, and filtering and scrolling run in different order. Maybe `scrollToEnd_filter_rowsUpdated`?

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java [hunk 2]
```diff
@@ -693,6 +693,23 @@ public void gridWithFiltering() {
...
+
+        // Scroll to the end of the grid
+        grid.scrollToRow(grid.getRowCount() - 1);
+        // Filter "Name" column with "100"
+        grid.findElement(By.tagName("vaadin-text-field")).sendKeys("100");
+        waitUntil(driver -> grid.getRowCount() == 1);
+
+        // Expect the one remaining row's first cell to contain
+        // text "Person 100"
+        Assert.assertEquals("Person 100", grid.getCell(0, 0).getText());
```
Comment (sissbruecker) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java:710]: When removing the fix, I'm getting a bit of a different actual value here compared to what is described in the issue. In the issue, and when testing manually, the cell seems to be empty. However when running the IT, it contains a value:
```
 GridViewIT.gridWithFiltering_scrolledToEnd:710 expected:<Person [100]> but was:<Person [489]>
```
Apart from that I can confirm that the change fixes the issue, and that the test fails when removing the fix, so maybe this is OK.

**Review by sissbruecker — COMMENTED** [review]
Looks good, verified that it works with manual testing. 

I see that this only targets master, 22, and 21. Did you check if the bug is also in 14?

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java (hunk 2 above)
Comment (tomivirkki) [reply vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java:710]: scrollToEnd_filter_rowsUpdated ANY_Chrome_ (com vaadin flow component grid it GridViewIT)

In the error screenshot (as in the original issue), the row seems visually empty. The cells however do contain text as the test output shows, but the row is in `[part~='row'][loading]` state so the content has `opacity: 0`.

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java (hunk 1 above)
Comment (tomivirkki) [reply vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java]: Done

**Comment by tomivirkki** [comment]
> I see that this only targets master, 22, and 21. Did you check if the bug is also in 14?

The issue doesn't reproduce with V14. The `<vaadin-grid>` Web Component has diverged quite a lot between V14-V2*. The V14 grid connector most likely works in this case (by accident) because the older `<vaadin-grid>` makes excess update requests for each row. The new version [was optimized](https://github.com/vaadin/web-components/pull/2910) to make just one data request / index.

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java (hunk 2 above)
Comment (sissbruecker) [reply vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java:710]: Got it, thanks for the explanation.

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**